src/pages/en/guides/deploy/google-cloud.mdx

```diff
@@ -11,21 +11,80 @@ i18nReady: true
 
 Deploying a project to Google Cloud requires some UI clicks. (Most of these actions can also be done using the [gcloud CLI](https://cloud.google.com/sdk/gcloud/)).
 
-### Cloud Run
+### Cloud Run (SSR and static)
+
+Cloud Run is a serverless platform that allows you to run a container without having to manage any infrastructure. It can be used to deploy both static and SSR sites.
+
+#### 1. Preparing the service
 
 1. Create a new GCP project, or select one you already have.
 
 2. Make sure the Cloud Run API is enabled.
 
 3. Create a new service.
 
-4. Use a container from Docker Hub or build your own using [Cloud Build](https://cloud.google.com/build).
+4. Make an empty `Dockerfile` in the root of your project.
+
+#### 2. Building the container
+
+There are a couple of ways to build the container:
+
+- Locally using Docker
+- Using [Cloud Build](https://cloud.google.com/build)
+- Using any other CI/CD tool (e.g. [GitHub Actions](<https://docs.github.com/en/actions/publishing-packages/publishing-docker-images>))
+
+##### Dockerfile for static sites (example)
+
+This following Dockerfile will build your site and serve it using Apache HTTPD (default config) on port `80`.
+
+```docker title="Dockerfile"
+FROM node:lts as build
+WORKDIR /app
+COPY . .
+RUN npm i
+RUN npm run build
+
+FROM httpd:2.4 as runtime
+COPY --from=build /app/dist /usr/local/apache2/htdocs/
+EXPOSE 80
+```
+
+:::note[notes]
+Keep in mind you might need to change `node:lts` to a pinned version, if you're using another package manager than `npm`, you might need to install it and change the scripts.
```

Comment:
**"In what situations would we want to change node:lts to a pinned version?"**:

Well, a simple example could be; Imagine the Node LTS versions changing from 16 to 18 (not that long ago). If you just have `node:lts` set, you might experience breakage if Astro (or any other package) does not support that version yet

**"If we're using another package manager than npm, we just need to replace npm i and npm run build in this file, right?"**

Yes and no, depending on the image. For example, the standard `Node:lts` does include `yarn` & `npm`, but not `pnpm`. So if you want to use `npm`, you could do something like:

```docker
FROM node:lts as build
RUN npm i -g pnpm@latest
RUN pnpm i
RUN pnpm build
```

Reply:
I like including that example so it's clear what you mean.
Also, I believe we can leave out the `pinned version` point here because we make it in the following section.

````suggestion
If you're using a package manager other than `npm`, you'll need to change the commands:

```docker title="Dockerfile" add={4} "pnpm"
FROM node:lts as build
WORKDIR /app
COPY . .
RUN npm i -g pnpm@latest
RUN pnpm i
RUN pnpm run build
...
```

````